js/src/lib/components/InferenceWidget/shared/helpers.ts

```diff
@@ -3,30 +3,18 @@ import { randomItem, parseJSON } from "../../../utils/ViewUtils";
 import type { WidgetExample } from "./WidgetExample";
 import type { ModelLoadInfo, TableData } from "./types";
 
-const VALID_QUERY_PARAMS = [
-	// string vals
-	"text",
-	"context",
-	"question",
-	"query",
-	"candidateLabels",
-	// table vals
-	"multiClass",
-	"table",
-	// boolean vals
-	"structuredData",
-];
-export type QueryParam = typeof VALID_QUERY_PARAMS[number];
+type KeysOfUnion<T> = T extends any ? keyof T : never;
+export type QueryParam = KeysOfUnion<WidgetExample>;
 type QueryParamVal = string | null | boolean | (string | number)[][];
 export function getQueryParamVal(key: QueryParam): QueryParamVal {
 	const searchParams = new URL(window.location.href).searchParams;
 	const value = searchParams.get(key);
-	if (["text", "context", "question", "query", "candidateLabels"].includes(key)) {
+	if (["text", "context", "question", "query", "candidate_labels"].includes(key)) {
```

Comment:
this will not work on existing URLs of ZeroShotClassification widget with URL query `candidateLabels` rather than `candidate_labels`. However, ZeroShotClassification widgets are not used widely. Therefore, I think this breaking change is fine

Reply:
I think i agree for  `candidate_labels` (it's used in the spec for widget examples) but for `structuredData` it's used that way in the spec no?

https://huggingface.co/docs/hub/models-widgets-examples#structured-data-classification

Reply:
wdyt #1064?
I can either merge #1064 or revert the changes back to using `strucutredData`. I will let you decide

Reply:
see my comment there